### PR TITLE
Expand AdHoc build workflow, add debug bookmarks screen

### PR DESCRIPTION
**Pull request by bwaresiak**
Task/Issue URL: https://app.asana.com/0/414235014887631/1207821981275160/f

https://app.asana.com/0/856498667320406/1207823615637251/f

**Description**:

Expand AdHoc build workflow to have an option of alpha and regular builds. Add debug bookmarks screen.

**Steps to test this PR**:

Check the workflow file for any obvious mistakes, validate history of executions for any mistakes.

Validate debug screen works as expected.

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: .github/workflows/adhoc.yml [hunk 1]
```diff
@@ -10,6 +10,14 @@ on:
...
         required: false
         type: string
+      is-alpha:
+        description: "Is alpha build"
+        type: choice
+        required: true
+        default: 'true'
+        options:
+           - true
+           - false
```
Comment (ayoy) [inline .github/workflows/adhoc.yml]: GHA supports boolean type so we can use it here. It would then use a checkbox in the UI, instead of a dropdown. The call site doesn't need updating:
```suggestion
      is-alpha:
        description: "Whether the build uses Alpha or Release configuration"
        type: boolean
        required: true
        default: true
```

Comment (bwaresiak) [reply .github/workflows/adhoc.yml]: I know, but I've seen people complaining that sometimes these behave like booleans, sometimes like strings, there's a whole thread https://github.com/actions/runner/issues/1483 so I'd prefer to explicitly use strings.

Comment (ayoy) [reply .github/workflows/adhoc.yml]: Yes, that's why the call site still needs to treat them as strings :) just pointing it out because it's possible to use a checkbox here, although on a second thought we could make it a dropbox with more descriptive options, like "Configuration: Alpha/Release". I'm not strong on any of my suggestions though, as long as the message is clear.

**Review by samsymons — COMMENTED** [review]
@bwaresiak When testing the debug menu I saved a couple favorites, but the debug menu doesn't show their metadata - it does show the root folders correctly.

Bookmarks

Then when deleting all bookmarks and re-running the app, I ran into this assertion failure:

Assertion Failure

A second run of the app worked, and I got the root folders recreated, but I still wasn't getting bookmark metadata in the debug menu.

File: DuckDuckGo/AppDelegate.swift [hunk 2]
```diff
@@ -117,6 +117,8 @@ import WebKit
     // swiftlint:disable:next function_body_length
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
 
+        OSLog.enabledLoggingCategories = ["Content Blocking"]
+
```
Comment (samsymons) [inline DuckDuckGo/AppDelegate.swift]: Is this change intentional for this PR?

Comment (bwaresiak) [reply DuckDuckGo/AppDelegate.swift]: No, I've missed this, thanks

File: .github/workflows/adhoc.yml (hunk 1 above)
Comment (bwaresiak) [reply .github/workflows/adhoc.yml]: I've tweaked the options and description a bit.

**Comment by bwaresiak** [comment]
@samsymons thanks, I've fixed the context (as it was being deallocated, making entries lose the content) and removed the delete option - since bookmarks can be managed by sync we don't want to just wipe them from db.

File: .github/workflows/adhoc.yml (hunk 1 above)
Comment (ayoy) [reply .github/workflows/adhoc.yml]: Thanks, that looks much clearer. Sorry to be a pain, but I'd also rename `is-alpha` now that it's not a boolean anymore. How about `build-type`? `configuration` is the most correct, but it also is a broad term in general and may be confusing.

Comment (bwaresiak) [reply .github/workflows/adhoc.yml]: Good suggestion, done :)

**Review by samsymons — APPROVED** [review]
LGTM!

**Review by ayoy — APPROVED** [review]
LGTM :)

**Comment by bwaresiak** [comment]
Thank you!
